### PR TITLE
[MS365] [Active Users] Add Active Users datastream

### PR DESCRIPTION
**Pull request by lucian-ioan**
## Proposed commit message

Add Active Users datastream.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/o365_metrics/data_stream/active_users/fields/fields.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+- name: o365.metrics.active_users
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/active_users/fields/fields.yml:1]: shall we follow naming as o365.metrics.active.users ?

File: packages/o365_metrics/data_stream/active_users/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        required: true
+        default: 3m
+      - name: processors
+        type: yaml
+        title: Processors
+        multi: false
+        required: false
+        show_user: false
+        description: >
+          Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/active_users/manifest.yml:24]: we can use this link for Processors: https://www.elastic.co/guide/en/fleet/current/elastic-agent-processor-configuration.html

File: packages/o365_metrics/data_stream/outlook_activity/fields/fields.yml [hunk 3]
```diff
@@ -19,8 +19,7 @@
         The count of email messages received by users during the reporting period.
     - name: emails_sent.count
       type: integer
-      description: |
-        The count of email messages sent by users during the reporting period.    
+      description: "The count of email messages sent by users during the reporting period.    \n"
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/outlook_activity/fields/fields.yml:22]:
```suggestion
      description: "The count of email messages sent by users during the reporting period."
```

File: packages/o365_metrics/changelog.yml [hunk 4]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.2.0"
```
Comment (ritalwar) [inline packages/o365_metrics/changelog.yml:2]:
```suggestion
- version: "0.1.1"
```

File: packages/o365_metrics/changelog.yml [hunk 5]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.1.1"
+  changes:
+    - description: Add `active_users` data stream.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/11937
```
Comment (andrewkroh) [inline packages/o365_metrics/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/11934
```

File: packages/o365_metrics/data_stream/active_users/fields/fields.yml [hunk 6]
```diff
@@ -0,0 +1,67 @@
+- name: o365.metrics.active.users
+  type: group
+  fields:
+    - name: exchange.active
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/active_users/fields/fields.yml:4]: Can we add the unit suffix "count" wherever applicable to the fields in this stream?

File: packages/o365_metrics/data_stream/active_users/fields/fields.yml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        Number of OneDrive inactive users.
+    - name: sharepoint.active
+      type: integer
+      description: |
+        Number of SharePoint active users.
+    - name: sharepoint.inactive
+      type: integer
+      description: |
+        Number of SharePoint inactive users.
+    - name: skype_for_business.active
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/active_users/fields/fields.yml:36]: The data for the `skype_for_business` field is not listed among the required metrics for this API. We can remove it for both active and inactive counts.

File: packages/o365_metrics/data_stream/outlook_activity/manifest.yml [hunk 8]
```diff
@@ -22,11 +22,13 @@ streams:
         show_user: false
         description: >
           Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/outlook_activity/manifest.yml:24]: Use agent link for processors

File: packages/o365_metrics/data_stream/active_users/manifest.yml [hunk 9]
```diff
@@ -0,0 +1,48 @@
+title: Microsoft Office 365 Active Users metrics
+type: metrics
+streams:
+  - input: cel
+    title: Office 365 Active Users metrics.
+    enabled: true
+    description: Collect Office 365 Active Users metrics via the Microsoft Graph API using CEL Input.
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/active_users/manifest.yml:7]:
```suggestion
    description: Collect Office 365 Active Users metrics.
```

File: packages/o365_metrics/_dev/build/docs/README.md [hunk 10]
```diff
@@ -43,4 +43,8 @@ Uses the Office 365 Management Graph API to retrieve metrics from Office 365.
 
 
 ### One Drive Usage
-{{fields "onedrive_usage"}}
\ No newline at end of file
+{{fields "onedrive_usage"}}
+
+
+### Active Users
+{{fields "active_users"}}
```
Comment (ritalwar) [inline packages/o365_metrics/_dev/build/docs/README.md:50]: Can you please also add sample_event.json and refer the event here? I recently did it for other datastreams in this [PR](https://github.com/elastic/integrations/pull/11687).

File: packages/o365_metrics/data_stream/active_users/elasticsearch/ingest_pipeline/default.yml [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+      field: active_users
+      target_field: o365.metrics.active.count.users   
+  - script:
+      lang: painless
+      description: Replace spaces and dashes in field names under o365.metrics.active.count.users.
+      tag: painless_purge_spaces_and_dashes
+      if: ctx.o365.metrics?.active?.count?.users instanceof Map
+      source: |
+        String underscore(String s) {
+          return /[ -]/.matcher(s).replaceAll('_').toLowerCase();
```
Comment (ritalwar) [inline packages/o365_metrics/data_stream/active_users/elasticsearch/ingest_pipeline/default.yml:17]: We need to update this, as the JSON in Elasticsearch contains a BOM character, which needs to be handled.
```suggestion
          String result = /[ -]/.matcher(s).replaceAll('_').toLowerCase();
          return /[\ufeff]/.matcher(result).replaceAll('')
```

**Review by ritalwar — APPROVED** [review]
LGTM!

**Pull request merged**